test/lib/dependency-tree/index.spec.ts

```diff
@@ -0,0 +1,78 @@
+import { AnalyzedPackageWithVersion } from "../../../lib/analyzer/types";
+import { buildTree } from "../../../lib/dependency-tree";
+
+const targetImage = "snyk/deptree-test@1.0.0";
+const targetOS = {
+  name: "snykos",
+  version: "1",
+  prettyName: "SnykOS GNU/Linux 1",
+};
+const freqDep = {
+  Name: `libsut`,
+  Version: "1.0",
+  Provides: [],
+  Deps: {},
+  AutoInstalled: true,
+  Purl: `pkg:snyk/libsut@1.0`,
+};
+
+function buildFreqDepsList(count: number) {
+  const depsInfoList: AnalyzedPackageWithVersion[] = [];
+
+  for (let i = 0, l = count; i < l; i = i + 1) {
+    depsInfoList.push({
+      Name: `libtest-${i + 1}`,
+      Version: "1.0",
+      Provides: [],
+      Deps: { libsut: true },
+      AutoInstalled: true,
+      Purl: `pkg:snyk/libtest-${i + 1}@1.0`,
+    });
+  }
+
+  depsInfoList.push(freqDep);
+
+  return depsInfoList;
+}
+
+function obj2array(obj) {
+  const arr: any[] = [];
+  for (const dep in obj) {
+    if (obj.hasOwnProperty(dep)) {
+      arr.push(obj[dep]);
+    }
+  }
+
+  return arr;
+}
+
+describe("dependency-tree", () => {
+  describe("buildTree", () => {
+    describe("Linux Package Managers", () => {
+      it("should attach frequent deps to parent when threshold is exceeded", () => {
+        const fixture = buildFreqDepsList(100);
+        const tree = buildTree(targetImage, "deb", fixture, targetOS);
+        const res = obj2array(tree.dependencies);
+        expect(
+          res.filter((dep) => !!dep.dependencies[freqDep.Name]),
```

Comment:
First time I've seen `!!` being used. I think it's okay, just letting you know it's not super clear.
Instead, I would have used `dep.dependencies.includes(freqDep.Name)`.

Reply:
Thanks @SteveShani I think this was left over from my initial tests that were coercing a `"true"` value to `true` where using `!!` seemed appropriate. This, however, is indeed confusing!

Reply:
Yes, I would use `hasOwnProperty` to check the scope of a given property to ensure it has not been inherited or within a `for in` loop to ensure I filter out built-ins (or avoid `for in` loops altogether 😄), but feel if I am simply checking for a known property in an object then this type of check is fine and clear on intent WDYT?

Reply:
Yeah definitely, whatever you choose to do is fine with me.